odoo/cli/obfuscate.py

```diff
@@ -0,0 +1,246 @@
+
+    def check_field(self, table, field):
+        qry = "SELECT udt_name FROM information_schema.columns WHERE table_name=%s AND column_name=%s"
+        self.cr.execute(qry, [table, field])
+        if self.cr.rowcount == 1:
+            res = self.cr.fetchone()
+            if res[0] in ['text', 'varchar']:
+                # Doesn t work for selection fields ...
+                return 'string'
+            if res[0] == 'jsonb':
+                return 'json'
+        return False
+
+    def get_all_fields(self):
+        qry = "SELECT table_name, column_name FROM information_schema.columns WHERE table_schema='public' AND udt_name IN ['text', 'varchar', 'jsonb'] AND NOT table_name LIKE 'ir_%' ORDER BY 1,2"
+        self.cr.execute(qry)
+        return self.cr.fetchall()
+
+    def convert_table(self, table, fields, pwd, with_commit=False, unobfuscate=False):
+        cypherings = []
+        cyph_fct = self.uncypher_string if unobfuscate else self.cypher_string
+
+        for field in fields:
+            field_type = self.check_field(table, field)
+            if field_type == 'string':
+                cypher_query = cyph_fct(sql.Identifier(field))
+                cypherings.append(sql.SQL('{field}={cypher}').format(field=sql.Identifier(field), cypher=cypher_query))
+            elif field_type == 'json':
+                # List every key
+                # Loop on keys
+                # Nest the jsonb_set calls to update all values at once
+                # Do not create the key in json if doesn't esist
+                new_field_value = sql.Identifier(field)
+                self.cr.execute(sql.SQL('select distinct jsonb_object_keys({field}) as key from {table}').format(field=sql.Identifier(field), table=sql.Identifier(table)))
+                keys = [k[0] for k in self.cr.fetchall()]
+                for key in keys:
+                    cypher_query = cyph_fct(sql.SQL("{field}->>{key}").format(field=sql.Identifier(field), key=sql.Literal(key)))
+                    new_field_value = sql.SQL("""jsonb_set({new_field_value}, array[{key}], to_jsonb({cypher_query})::jsonb, FALSE) """).format(new_field_value=new_field_value, key=sql.Literal(key), cypher_query=cypher_query)
+                cypherings.append(sql.SQL('{field}={cypher}').format(field=sql.Identifier(field), cypher=new_field_value))
+
+        if cypherings:
+            query = sql.SQL("UPDATE {table} SET {fields}").format(table=sql.Identifier(table), fields=sql.SQL(',').join(cypherings))
+            self.cr.execute(query, {"pwd": pwd})
+            if with_commit:
+                self.commit()
+                self.begin()
+
+    def confirm_not_secure(self):
+        _logger.info("The obfuscate method is not considered as safe to transfer anonymous datas to a third party.")
+        conf_y = input(f"This will alter data in the database {self.db_name} and can lead to a data loss. Would you like to proceed [y/N]? ")
```

Comment:
```suggestion
        conf_y = input(f"This will alter data in the database {self.dbname} and can lead to a data loss. Would you like to proceed [y/N]? ")
```